timemachine/cpp/src/local_md_potentials.cu

```diff
@@ -190,18 +214,20 @@ void LocalMDPotentials::_setup_free_idxs_given_reference_idx(
 
     // The row indices is all of the free indices, which excludes the reference
     const int num_row_idxs = p_num_selected_.data[0];
-    // The col indices is all indices, except for the free, including the reference which is frozen.
-    const int num_col_idxs = N_ - num_row_idxs;
+    const int num_col_idxs = N_ - num_row_idxs - indices_to_remove;
 
     if (num_row_idxs == 0) {
         throw std::runtime_error("LocalMDPotentials setup has no free particles selected");
     }
 
-    // The reference particle will always be in the column idxs
-    if (num_row_idxs == N_ - 1) {
+    // The reference particle will always be in the column idxs if the referenc is frozen
```

Comment:
typo: reference

Reply:
Fixed in https://github.com/proteneer/timemachine/pull/1038/commits/7dbeabcffb9e5b0cc94b1ccebbf69ebaf8eefb1a